### ISSUE TITLE
Allow for playback of only a specific section of an animation

### ISSUE THREAD

**Issue by QbieShay** [godotengine/godot-proposals#8526]
### Describe the project you are working on

VFX - specifically a long cinematic sequence with multiple VFX

### Describe the problem or limitation you are having in your project

I am using the animation player to drive multiple VFX. Since this is a small cinematic sequence, I need to animate everything together on the same timeline to make sure timings fit with one another.

There's a couple of issues right now with this workflow, but the biggest one for me right now is that I can't play and replay only a section of an animation.

### Describe the feature / enhancement and how it helps to overcome the problem or limitation

Allow for selecting a section of an animation for playback (editor only is fine)

### Describe how your proposal will work, with code, pseudo-code, mock-ups, and/or diagrams

Add a cntrol at the bottom of the animation player to optionally clamp the playback range:
one icon, two boxes for the clamp.

![image](https://github.com/godotengine/godot-proposals/assets/7917475/732454e6-f05c-45bf-8c82-ed9282b83347)

### If this enhancement will not be used often, can it be worked around with a few lines of script?

It's the animation player

### Is there a reason why this should be core and not an add-on in the asset library?

/

**Comment by Calinou** [comment]
For reference, the workflow for doing so in Audacity 3.3.3 is as follows:

- There's a Loop button that toggles loop-on-playback behavior with an adjustable section. Here it's disabled, indicated by a yellow loop section:

![image](https://github.com/godotengine/godot-proposals/assets/180032/7c8f0011-1315-462a-8ea3-ed94b5558f59)

- When Loop is enabled, the loop section on the timeline turns white:

![image](https://github.com/godotengine/godot-proposals/assets/180032/9e358697-c1b3-4783-9377-688d7a5d91ac)

You can change the loop section by dragging over the timeline with the left mouse button.

- If you start playback outside the loop section, it's ignored until the loop section is reached:

https://github.com/godotengine/godot-proposals/assets/180032/42581bab-43cd-4c60-b92e-a13b890f8b22

This allows previewing an animation from the start and viewing a looping section only once it's reached.

**Comment by TokageItLab** [comment]
The restrictions on how to obtain Audio/AnimationPlayback keys described here https://github.com/godotengine/godot/issues/85088#issuecomment-1826846802 are relevant a bit. I keep them and this proposal in mind when working later.

**Comment by QbieShay** [comment]
Thank you for looking at it @TokageItLab ! If this is something you're already working on, would it be accurate to move the state of this proposal to "in progress" and assign you to it?

**Comment by TokageItLab** [comment]
I just indicated that the problem I linked to is a regression in a sense, and that it has simply become this proposal blocker. So unfortunately, I would not be able to work on this at the same time.

After that fix, I think this proposal will be technically possible, but it would be better to work on it at the same time as the A-B Repeat/Loop proposal, which is also available at run-time; I think someone wanted that somewhere, but I can't recall.
